graylog2-server/src/main/java/org/graylog2/lookup/db/LookupStorage.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.lookup.db;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.DB;
+import com.mongodb.DBCollection;
+import com.mongodb.WriteConcern;
+import org.bson.types.ObjectId;
+import org.graylog2.bindings.providers.MongoJackObjectMapperProvider;
+import org.graylog2.database.MongoConnection;
+import org.graylog2.lookup.dto.LookupStorageDto;
+import org.graylog2.plugin.Tools;
+import org.joda.time.DateTime;
+import org.mongojack.DBQuery;
+import org.mongojack.JacksonDBCollection;
+import org.mongojack.WriteResult;
+
+import javax.inject.Inject;
+
+import java.util.Optional;
+
+import static org.graylog2.lookup.dto.LookupStorageDto.FIELD_LOOKUP_KEY;
+import static org.graylog2.lookup.dto.LookupStorageDto.FIELD_UPDATED_AT;
+
+/**
+ * This class provides MongoDB-backed storage for data retrieved by Data Adapters.  Each Data Adapter's data will
+ * be stored in a unique MongoDB collection named "lut_data_adapter_storage_Data-Adapter-ID".  The init() method should
```

Comment:
Where do you see the benefit of creating a new collection for each data adapter instead of using a single table?

Reply:
While it shouldn't make any real difference in query speed, I thought it would make things easier if anything went wrong and a human had to look at what was going on in MongoDB.

Reply:
@waab76 Thanks, sounds good! I read up on that topic [here](https://docs.mongodb.com/manual/core/data-model-operations/#large-number-of-collections) as well and there doesn't seem to be any issues with large number of collections.

If we have one collection per data adapter we can even use a [TTL index](https://docs.mongodb.com/manual/tutorial/expire-data/) to expire data automatically.